mod_app/middleware.py

```diff
@@ -0,0 +1,35 @@
+from django.contrib import admin
+from django.contrib.sites.shortcuts import get_current_site
+
+
+def admin_custom_titles(request):
+    current_site = get_current_site(request)
+    site_header = "Administration for Museum of Dreamworlds Site"
+    site_title = "Museum of Dreamworlds Admin Site"
+    index_title = "Admin"
+
+    if "museumofdreamworlds" not in current_site.domain:
```

Comment:
Could you do this by querying settings by specifying the deployment environment there? e.g.

```
from django.conf import settings

if settings.DEPLOYMENT_ENVIRONMENT == "production":
    site_header = "Administration for Museum of Dreamworlds Site"
else:
   site_header = "Administration for Museum of Dreams (staging)"
```
Would require separate staging and production settings files, but that's probably not a bad thing